etl/src/main/java/org/pih/hivmigration/etl/sql/LabResultMigrator.groovy

```diff
@@ -0,0 +1,261 @@
+            SELECT r.obs_id, p.person_id, @cd4_construct, e.encounter_id, r.obs_datetime, e.location_id, 1, now(), 0, r.uuid
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='cd4';
+            
+            -- Hematocrit
+            SET @hematocrit_construct = (SELECT concept_id FROM concept WHERE uuid='267C165C-1B8F-48FE-91AC-C1AE8C7412A0');
+            INSERT INTO obs(obs_id, person_id, concept_id, encounter_id, obs_datetime, location_id, creator, date_created, voided, uuid)
+            SELECT r.obs_id, p.person_id, @hematocrit_construct, e.encounter_id, r.obs_datetime, e.location_id, 1, now(), 0, r.uuid
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='hematocrit';
+            
+            -- PPD
+            SET @ppd_construct = (SELECT concept_id FROM concept WHERE uuid='ACF90DED-B595-4356-9840-788094C60AFB');
+            INSERT INTO obs(obs_id, person_id, concept_id, encounter_id, obs_datetime, location_id, creator, date_created, voided, uuid)
+            SELECT r.obs_id, p.person_id, @ppd_construct, e.encounter_id, r.obs_datetime, e.location_id, 1, now(), 0, r.uuid
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='ppd';
+            
+            -- Rapid test
+            SET @test_rapid_construct = (SELECT concept_id FROM concept WHERE uuid='A19F5A83-E960-413D-B93B-9270C53580A2');
+            INSERT INTO obs(obs_id, person_id, concept_id, encounter_id, obs_datetime, location_id, creator, date_created, voided, uuid)
+            SELECT r.obs_id, p.person_id, @test_rapid_construct, e.encounter_id, r.obs_datetime, e.location_id, 1, now(), 0, r.uuid
+            FROM hivmigration_lab_results r
+                     JOIN hivmigration_patients p ON r.source_patient_id = p.source_patient_id
+                     JOIN hivmigration_encounters e ON r.source_encounter_id = e.source_encounter_id
+            WHERE r.test_type ='tr';
+            
+            --
+            -- Fill in the constructs
+            --
+            -- Now for each construct we create the expected members.
+            
+            SET @specimen_number = (SELECT concept_id FROM concept WHERE uuid='162086AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+            
+            -- Viral Load
+            --
+            
+            -- Add specimen number
```

Comment:
It seems unnecessary to repeat this population of specimen number for each test type individually.  This same query is repeated for each.  Why do that?  Furthermore, I suspect that this will result in the creation of obs with null values, since this doesn't do any checking that specimen number is actually populated.

I'd remove all of the individual queries, and just have a single section for "migrate specimen numbers", which migrates everything from hivmigration_lab_results where sample_id is not null.

Reply:
Following on my comment about constructs above, in the Lab Workflow module the specimen number would be a top-level property of the encounter, not of the individual tests, and this would allow us to avoid creating constructs for all these results.

Tangent: we should have a Confluence page (if we don't already) about our lab results modellign strategy.

Reply:
@mogoodrich yeah, agreed.  Ellen also isn't aware of any reason we'd need the constructs with specimen numbers. @mseaton any reason I shouldn't get rid of them?

Reply:
@brandones - are you implying that we should just discard the sample ids and not migrate them at all?

Reply:
@mseaton  No, I'm saying that based on what I've heard from Mark and Ellen, that we probably don't need to create these constructs with sample ids migrated as specimen numbers. We can do anything we like with the sample ids. I think Mark's suggestion sounds reasonable.

Reply:
I see @brandones , yeah that makes sense.  Sorry, I didn't see @mogoodrich comments.